src/main/java/org/candlepin/subscriptions/subscription/export/SubscriptionDataExporterService.java

```diff
@@ -122,63 +131,165 @@ private DbReportCriteria extractExportFilter(ExportServiceRequest request) {
       }
     }
 
-    return report.build();
+    return criteria;
   }
 
-  private static void handleProductIdFilter(
-      DbReportCriteria.DbReportCriteriaBuilder builder, String value) {
+  private static Specification<SubscriptionCapacityView> handleProductIdFilter(String value) {
     var productId = ProductId.fromString(value).getValue();
-    if (SubscriptionDefinition.isPrometheusEnabled(productId)) {
-      builder.productTag(productId);
-    } else {
-      builder.productId(productId);
-    }
+    return (root, query, builder) ->
+        builder.equal(root.get(SubscriptionCapacityView_.productTag), productId);
   }
 
-  private static void handleUsageFilter(
-      DbReportCriteria.DbReportCriteriaBuilder builder, String value) {
+  private static Specification<SubscriptionCapacityView> handleUsageFilter(String value) {
     Usage usage = Usage.fromString(value);
     if (value.equalsIgnoreCase(usage.getValue())) {
-      builder.usage(usage);
+      if (!Usage._ANY.equals(usage)) {
+        return (root, query, builder) ->
+            builder.equal(root.get(SubscriptionCapacityView_.usage), usage.getValue());
+      }
     } else {
       throw new IllegalArgumentException(String.format("usage: %s not supported", value));
     }
+
+    return null;
   }
 
-  private static void handleCategoryFilter(
-      DbReportCriteria.DbReportCriteriaBuilder builder, String value) {
-    builder.hypervisorReportCategory(
-        HypervisorReportCategory.mapCategory(ReportCategory.fromString(value)));
+  private static Specification<SubscriptionCapacityView> handleCategoryFilter(String value) {
+    String measurementType = getMeasurementTypeFromCategory(value);
+    if (measurementType != null) {
+      return (root, query, builder) ->
+          builder.like(
+              builder.upper(builder.function("jsonb_pretty", String.class, root.get("metrics"))),
+              "%" + measurementType.toUpperCase(Locale.ROOT) + "%");
```

Comment:
Could we do these `like` expressions using Postgresql's native JSON operations?  Pretty-printing the JSON and then doing a `LIKE` on the results seems potentially error-prone and likely inefficient.

```sql
select * from subscription_capacity_view where metrics @? '$[*] ? (@.measurement_type == "PHYSICAL")';
```

uses the Postgresql [JSON path](https://www.postgresqltutorial.com/postgresql-json-functions/postgresql-json-path/) syntax.

I realize that hammering that into the criteria query API isn't easy, but maybe you can call out to the `jsonb_path_exists` function?

```sql
select * from subscription_capacity_view where jsonb_path_exists(metrics, '$[*] ? (@.measurement_type == "PHYSICAL")')
```

I also recognize that this might complicate testing on HSQLDB but I don't think HSQLDB supports `jsonb_pretty` either so we shouldn't be in a worse position using the Postgresql JSON path functions.

Reply:
I'm not familiar with `jsonb_path_exists` but it seems to match much better. Will try to use it and update the PR accordingly afterwards. Thanks!

Reply:
@awood PR updated including this change.